woocommerce/woocommerce
# [Settings]: update @wordpress/edit-site dependency to 6.14.0

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Currently, using WooCommerce Settings with the Gutenberg development branch is not possible.
The `@wordpress/edit-site` API has changed considerably, introducing breaking ones.

This PR updates the package's dependency and relevant changes to make it work:

### `routes`

This PR introduces the settings routes used to decorate the different parts of the layout in the same way the core does.
It uses the `registerRoute` private action, and also the `getRoutes` private selector.

### `Sidebar`

* It uses the `SidebarNavigationItem` core component to render the sidebar
* It removes the custom `SettingsItem` component e76317fbb94b578ccbd176d9b811a8109d0c73f4

### Styles

It uses core mixings to reset part of the layout

Closes https://github.com/woocommerce/woocommerce/issues/53818



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Update the dependencies and build the app
2. Confirm it's possible to change the Settings section by clicking on the sidebar

https://github.com/user-attachments/assets/75a2c49f-b07b-4b1e-8659-156270829e71




### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
